internal/controller/state.go

```diff
@@ -68,6 +68,45 @@ func (clctrl *ClusterController) StateStoreCredentials() error {
 				telemetry.SendEvent(clctrl.TelemetryEvent, telemetry.StateStoreCredentialsCreateFailed, err.Error())
 				return fmt.Errorf("failed to update cluster after creating AWS state store: %w", err)
 			}
+		case "azure":
+			// Azure storage is non-S3 compliant
+			location := "eastus"               // @todo(sje): allow this to be configured
```

Comment:
I've had an idea how to do this - will add something to remove the `@todos` tomorrow morning